### PR TITLE
Update alias params

### PR DESCRIPTION
**Pull request by kainkad**
- updated params with their documented names not aliases
- update default values to match lightgbm [documentation](https://lightgbm.readthedocs.io/en/v3.3.5/Parameters.html)
- re-enabled formerly broken tests
- relates to https://github.com/IQVIA-ML/LightGBM.jl/issues/23#issuecomment-614446682 (but leaves the issue opened due to other discussions)

### CODE REVIEW

File: src/estimators.jl [hunk 1]
```diff
@@ -83,14 +83,14 @@ end
     LGBMRegression(; [
         objective = "regression",
         boosting = "gbdt",
-        num_iterations = 10,
```
Comment (kainkad) [inline src/estimators.jl:86]: in few cases the default parameter values differed to the official documentation defaults without any specific reason so I updated them accordingly given that all other parameters have defaults according to lightgbm documentation

File: src/estimators.jl [hunk 2]
```diff
@@ -701,9 +701,9 @@ function LGBMRanking(;
     max_cat_threshold = 32,
     cat_l2 = 10.0,
     cat_smooth = 10.0,
-    metric = [""],
+    metric = ["ndcg"],
```
Comment (kainkad) [inline src/estimators.jl:704]: Updated this default as the default objective for this estimator is lambdarank and these two usually go together. Unlike for binary, multiclass or regression objectives, where the empty "" metric is filled with default value corresponding to the pre-defined objective, the empty string doesn't assign the metric in this case.

File: src/estimators.jl [hunk 3]
```diff
@@ -61,13 +60,13 @@ mutable struct LGBMRegression <: LGBMEstimator
...
+    is_provide_training_metric::Bool
     eval_at::Vector{Int}
 
     num_machines::Int
     local_listen_port::Int
     time_out::Int
-    machine_list_file::String
+    machine_list_filename::String
 
     num_class::Int
```
Comment (kainkad) [inline src/estimators.jl:71]: `num_class` shouldn't be a parameter for the regressor but it's hardcoded to 1 and internally used to return reshaped predictions: https://github.com/IQVIA-ML/LightGBM.jl/blob/master/src/predict.jl#L41-L43 so removing it would require to find some decent equivalent in returning these predictions

File: src/MLJInterface.jl [hunk 4]
```diff
@@ -82,15 +81,15 @@ MLJModelInterface.@mlj_model mutable struct LGBMRegressor <: MLJModelInterface.D
     # Metrics
     metric::Vector{String} = ["l2"]::(all(in.(_, (LGBM_METRICS, ))))
```
Comment (kainkad) [inline src/MLJInterface.jl:82]: Unfortunately defaulted values are required on the MLJInterface so here ["l2"] has been left where it is [""] in estimators.jl (which means a default for pre-defined objectives).
This applies to all the cases where we could use an empty string. 
image

File: src/estimators.jl [hunk 5]
```diff
@@ -83,14 +82,14 @@ end
...
         boosting = "gbdt",
-        num_iterations = 10,
+        num_iterations = 100,
         learning_rate = .1,
-        num_leaves = 127,
+        num_leaves = 31,
         max_depth = -1,
         tree_learner = \"serial\",
-        num_threads = Sys.CPU_THREADS,
+        num_threads = 0,
```
Comment (kainkad) [inline src/estimators.jl:90]: as per documentation the real cpu cores achieve better speed but the actual default is 0 so changed it across all https://lightgbm.readthedocs.io/en/v3.3.5/Parameters.html#num_threads

File: src/estimators.jl [hunk 6]
```diff
@@ -371,14 +366,14 @@ end
         max_cat_threshold = 32,
         cat_l2 = 10.0,
         cat_smooth = 10.0,
-        metric = [\"multi_logloss\"],
+        metric = [""],
```
Comment (kainkad) [inline src/estimators.jl:369]: Default for metric is an empty string `""` not a vector but this applies to one supported metric, or metrics introduced as a string such as `metric=binary_logloss` whereas for most use cases with multiple metrics we need a vector of strings, hence leaving it here.

File: src/MLJInterface.jl [hunk 7]
```diff
@@ -155,7 +153,7 @@ MLJModelInterface.@mlj_model mutable struct LGBMClassifier <: MLJModelInterface.
     objective::String = "multiclass"::(_ in CLASSIFICATION_OBJECTIVES)
```
Comment (kainkad) [inline src/MLJInterface.jl:153]: Can't comment against the commented out sigmoid param hence commenting here (see lines above for sigmoid). I'm not quite sure about adding sigmoid here because of this: https://github.com/IQVIA-ML/LightGBM.jl/blob/master/src/MLJInterface.jl#L150-L152. Since the defaulted objective on the MLJ classifier is multiclass, then adding a defaulted sigmoid here might cause some issues but on the other hand the LGBMClassification also is defaulted to multiclass and still has the sigmoid defaulted to 1.

File: src/estimators.jl [hunk 8]
```diff
@@ -371,14 +366,14 @@ end
...
         metric_freq = 1,
-        is_training_metric = false,
+        is_provide_training_metric = false,
         eval_at = Int[1, 2, 3, 4, 5],
         num_machines = 1,
         local_listen_port = 12400,
         time_out = 120,
-        machine_list_file = \"\",
+        machine_list_filename = \"\",
         num_class = 2,
```
Comment (kainkad) [inline src/estimators.jl:377]: This does not match the default in the documentation but it' s because in the documentation also has the default objective set to regression for which num_class is also defaulted to 1.
In LighGBM.jl it's been decided to have now 3 estimators where the LGBMClassification is defaulted to multiclass (not binary) hence this default num_class.

File: src/MLJInterface.jl (hunk 7 above)
Comment (kainkad) [reply src/MLJInterface.jl:152]: Re-enabled `sigmoid` with a default value since we have several other parameters which work only with `binary` or `multiclassova` such as `scale_pos_weight` or `is_unbalance`. Their defaults have no impact with the currently default objective set to `multiclass` but are made available if the user changes the objective and wants to use them.

File: src/estimators.jl (hunk 8 above)
Comment (kainkad) [reply src/estimators.jl:377]: Just to clarify on this the regression task does not use num_class which is required for classification task. The default in this estimator differs from the default in the official docs because the classifier in LightGBM.jl is defaulted to multiclass objective.
